custom/icds_reports/tasks.py

```diff
@@ -1168,11 +1157,49 @@ def build_incentive_report(agg_date=None):
     state_ids = (SQLLocation.objects
                  .filter(domain=DASHBOARD_DOMAIN, location_type__name='state')
                  .values_list('location_id', flat=True))
+    locations = (SQLLocation.objects
+                 .filter(domain=DASHBOARD_DOMAIN, location_type__name__in=['state', 'district', 'block'])
+                 .select_related('parent__parent', 'location_type'))
     if agg_date is None:
         current_month = date.today().replace(day=1)
         agg_date = current_month - relativedelta(months=1)
     for state in state_ids:
         AWWIncentiveReport.aggregate(state, agg_date)
+    for file_format in ['xlsx', 'csv']:
+        for location in locations:
+            if location.location_type.name == 'state':
+                build_incentive_files.delay(location.location_id, agg_date, file_format, 1, location)
+            elif location.location_type.name == 'district':
+                build_incentive_files.delay(location, agg_date, file_format, 2, location.parent, location)
+            else:
+                build_incentive_files.delay(location, agg_date, file_format, 3, location.parent.parent, location.parent, location)
+
+
+@task(queue='icds_dashboard_reports_queue')
+def build_incentive_files(location, month, file_format, aggregation_level, state, district=None, beta=False):
+    data_type = 'AWW_Performance'
+    excel_data = IncentiveReport(
+        location=location.location_id,
+        month=month,
+        aggregation_level=aggregation_level,
+        beta=beta
+    ).get_excel_data()
+    state = state.name
+    district = district.name if aggregation_level >= 2 else None
```

Comment:
or `district_name = district.name if district else None`

Also FYI you are overriding the var names here for `state` and `district`